### DISCUSSION TITLE
Microsoft.AspNetCore.OData v8 support ?

### DISCUSSION THREAD

**Discussion by MDWide** [OData/RESTier#725] [Q&A] [answered]
Hello everyone,

I started to use the fantastic RESTier on a brand new ASP.NET Core 7 project.

When I select the Microsoft.Restier.AspNetCode nuget v 1.0.2 CI, its dependency on Microsoft.AspNetCore.OData says version >=7.6.4.

With this version, everything seems to work, but when I update the Microsoft.AspNetCore.OData nuget to version 8.1.1, suddenly some TypeLoadException arise (specifically Microsoft.AspNet.OData.Extensions.ODataServiceCollectionExtensions).

Is this 1.0.2 CI supposed to be compatible with Microsoft.AspNetCore.OData v8 or is it planned to ?

BR

**Comment by MDWide** [comment]
Hello,

So after a quick check on latest nuget package V1.1.0-CI, RESTier now seems to even have downgraded to 7.5.x version of Microsoft.AspNetCore.OData, which is kinda frightening.

AspNetCore Odata V8.0 has been released nearly two years ago.

We really need some communication from the RESTier team about the future of this framework.

Are there any plans to migrate to Odata V8 ?

BR

**Comment by robertmclaws** [comment, marked as answer]
OData v8 is a major rewrite from the OData architecture from v7, and as such requires significant work to upgrade. Moving to V8 is planned, but don't expect it in the near future.

In the meantime, v7 is still being updated and works just fine. You should use the version that Restier is pinned to, for compatibility reasons. Other OData releases have seen portions of the OData framework fail our unit tests, that's why we pinned it to a lower version. In my personal opinion, the 7.6 version should never have been released, it has had significant quality issues.

As an aside, if you are "frightened" by version changes, you might consider a different line of work. That's a little melodramatic.

**Comment by MDWide** [comment]
Hello,

Thank you for the clarification and taking the time to answer. It helped in making the decision to completely remove the OData/Restier stack from our project.

Maybe "frightening" wasn't the most appropriate term, but having confidence in the evolution of a technology/stack/framework is key when it comes to choose between one and another. 

And I understand your concerns about V7.6 and the work required to move to V8, yet I believe no one expects already lagging behind projects to go further backward on dependencies when new major version of them has been released years ago.

I blame Microsoft here for not supporting such awesome projects as they should.

BR
